### PR TITLE
rule: add support for dport/sport

### PR DESCRIPTION
**Pull request by chendotjs**
This PR adds ability equivalent to `ip rule add dport xxx-xxx sport xxx table main`

### CODE REVIEW

File: rule.go [hunk 1]
```diff
@@ -40,3 +42,14 @@ func NewRule() *Rule {
 		Flow:              -1,
 	}
 }
+
+// RulePortRange creates rule sport/dport range.
+func NewRulePortRange(start, end uint16) *RulePortRange {
```
Comment (aboch) [inline rule.go]: Why not making this a method of `Rule` structure ?
```
(r *Rule) AddPortRange(start, end uint16)
```

Comment (chendotjs) [reply rule.go]: Thanks for review @aboch 
`(r *Rule) AddPortRange(start, end uint16)` may need another parameter to specify it's a Src or a Dst port range, which might add more complexity.
IMO, current fields of `Rule` structure is assigned by users manually, so `NewRulePortRange` is enough.

**Comment by aboch** [comment]
@chendotjs pls rebase

**Comment by chendotjs** [comment]
@aboch done

**Comment by aboch** [comment]
LGTM

**Pull request merged**